sdk/go/common/workspace/plugins.go

```diff
@@ -533,17 +533,33 @@ func GetPluginDir() (string, error) {
 	return GetPulumiPath(PluginDir)
 }
 
-// GetPlugins returns a list of installed plugins.
+// GetPlugins returns a list of installed plugins without size info and last accessed metadata.
+// Plugin size requires recursively traversing the plugin directory, which can be extremely
+// expensive with the introduction of nodejs multilang components that have
+// deeply nested node_modules folders.
```

Comment:
I assume there is only one caller in the world that cares about this additional metadata (plugin ls). Can't we just modify that caller to call SetFileMetadata and just always skip filling in metadata here? (Instead of forking this API).

Reply:
While that is possible, what you're suggesting is to basically duplicate `workspace.getPlugins` in `cmd/plugin`. The `PluginInfo` returned does not contain the path of the plugin, which is required for `SetFileMetadata`. Doesn't really feel better than having this "fork". 

I'll leave as is for now. Like you said plugin_ls is the only consumer of this surface area, all other callers are doing the right thing, and we maintain the option to refactor later if it makes sense.